xapian-applications/omega/index_file.cc

```diff
@@ -663,7 +663,9 @@ index_mimetype(const string & file, const string & urlterm, const string & url,
 	    Worker* wrk = wrk_it->second;
 	    if (!wrk ||
 		!wrk->extract(file, dump, title, keywords, author, pages)) {
-		string msg = "Couldn't extract text from " + file;
+		string msg = wrk->get_error();
+		if (msg.empty())
+		    msg = "Couldn't extract text from " + file;
```

Comment:
Your documentation for `get_error()` says that if it returns an empty string that means "no error", in which case this handling this case isn't necessary (you could reasonably `assert(!msg.empty());` here).

Reply:
I am going to fix it.